Refactor/optimising posts

1MeghPatel1 opened the pull request.

1MeghPatel1 committed:
add (Bulk Comments Update with Rollback):
- added states and updated useComments hook to handle bulk comment updates and rollback in case of error

1MeghPatel1 committed:
add (Bulk Comments Update with Rollback):
- added states and updated useComments hook to handle bulk comment updates and rollback in case of error

thekalamanchofficial commented on src/app/(with-sidebar)/myfeed/context/FeedContext.tsx:
```diff
@@ -52,27 +103,58 @@ export const FeedProvider: React.FC<{ children: React.ReactNode }> = ({
+
+      await bulkCommentMutation.mutateAsync(input);
+    },
+    5000,
```
store this constant in common config file

thekalamanchofficial commented on src/app/(with-sidebar)/myfeed/context/FeedContext.tsx:
```diff
@@ -43,6 +70,30 @@ export const FeedProvider: React.FC<{ children: React.ReactNode }> = ({
 
+  const bulkCommentMutation = trpc.comments.bulkAddComments.useMutation({
+    onSuccess: (response) => {
+      console.log("BULK COMMENT RESPONSE", response);
```
remove log

thekalamanchofficial commented on src/app/(with-sidebar)/myfeed/context/FeedContext.tsx:
```diff
@@ -52,27 +103,58 @@ export const FeedProvider: React.FC<{ children: React.ReactNode }> = ({
     await bulkLikeMutation.mutateAsync(input);
   }, BULK_LIKE_DEBOUNCE_DELAY);
 
+  const debouncedSendComments = useDebounce(
```
try to replace with lodash debounce if that doesnt take too much time.

thekalamanchofficial commented on src/hooks/useComments.ts:
```diff
@@ -11,160 +12,210 @@ type UseCommentsProps = {
-        prev.filter(update => update.tempId !== tempComment.id)
+    });
+
+    // Todo: only remove failed comment from batch
```
remove todo

thekalamanchofficial commented on src/server/routers/comment.ts:
```diff
@@ -53,6 +71,145 @@ export const commentRouter = router({
+
+        const parentComments: BulkCommentInput[] = [];
+        const childComments: BulkCommentInput[] = [];
+        const tempToRealIdMap = new Map<string, string>();
```
change all these with object.id

1MeghPatel1 committed:
Merge branch 'main' of https://github.com/thekalamanchofficial/kalamanch into refactor/optimising-posts

1MeghPatel1 committed:
remove & resolve
- removed ID Maping and used mongo's new ObjectID() to create id in the client side as well, and removed processed comments logic with it
- resolved PR comments
- added next/typescript in es lint config file

thekalamanchofficial commented on src/app/(with-sidebar)/myfeed/types/types.ts:
```diff
@@ -7,6 +7,7 @@ export type MenuItemList = {
 };
 export type Comment = {
   id: string;
+  tempId?: string;
```
do we need this tempId

thekalamanchofficial commented on src/app/(with-sidebar)/myfeed/types/types.ts:
```diff
@@ -18,6 +19,16 @@ export type Comment = {
 };
 
+export type CommentPayload = {
+  tempId: string;
```
we need this tempId

thekalamanchofficial commented on src/hooks/useComments.ts:
```diff
@@ -11,160 +14,130 @@ type UseCommentsProps = {
-    try {
-      const response = await commentMutation.mutateAsync({
+      addCommentToBatch({
+        tempId: tempComment.id,
```
why are we still using this?

thekalamanchofficial commented on src/server/routers/comment.ts:
```diff
@@ -53,6 +71,87 @@ export const commentRouter = router({
+        }));
+
+        if (parentCommentData.length > 0) {
+          console.log({
```
remove log

thekalamanchofficial commented on src/server/routers/comment.ts:
```diff
@@ -53,6 +71,87 @@ export const commentRouter = router({
+        }));
+
+        if (childCommentData.length > 0) {
+          console.log({
```
remove log

1MeghPatel1 committed:
resolve
- resolved PR comments for logs and setBulkCommentsState

thekalamanchofficial commented on src/server/routers/comment.ts:
```diff
@@ -23,15 +23,33 @@ const commentSchema = yup.object({
 });
 
+const bulkCommentSchema = commentSchema.shape({
+  tempId: yup.string().required(),
```
do we need this tempid

thekalamanchofficial commented on src/server/routers/comment.ts:
```diff
@@ -53,6 +71,81 @@ export const commentRouter = router({
+          userName: comment.userName,
+          userProfileImageUrl: comment.userProfileImageUrl,
+          parentId: comment.parentId,
+          id: comment.tempId,
```
it seems we are using tempId for sending the objectId. If thats the case then we should rename it to just id

1MeghPatel1 committed:
resolve (PR Comments):
- resolved comments for renaming the tempId to id

1MeghPatel1 committed:
edit (Comments and Likes):
- changed debounce time for both to 1 minute
